### ISSUE TITLE
redis.pub_sub().get_message() subscribe channel duplicated

### ISSUE THREAD

**Issue by penguinol** [NoneGG/aredis#80]
## Checklist

- Python 3.6
- Python parser
- asyncio event loop
- 1.1.2

## Steps to reproduce
```python
from aredis import StrictRedis
import asyncio

async def sub_redis():
    redis = StrictRedis('172.16.179.100', 6379)
    redis_sub = redis.pubsub()
    await redis_sub.subscribe('notify')
    while True:
        message = await redis_sub.get_message(False, 5)
        print(message)


if __name__ == '__main__':
    asyncio.ensure_future(sub_redis())
    asyncio.get_event_loop().run_forever()
```
## Expected behavior
with redis-py, it only call subscribe once, and receive subscribe notify once
> {'type': 'subscribe', 'pattern': None, 'channel': b'notify', 'data': 1}
> None
> None
> None

## Actual behavior
> {'type': 'subscribe', 'pattern': None, 'channel': b'notify', 'data': 1}
> None
> {'type': 'subscribe', 'pattern': None, 'channel': b'notify', 'data': 1}
> None
> {'type': 'subscribe', 'pattern': None, 'channel': b'notify', 'data': 1}

with aredis, it will receive subscribe notify every time when get_message() is called

**Comment by NoneGG** [comment]
@penguinol 
You can see subscription message because `timeout` is set to 5, and if in 5 seconds there are no messages published, the `TimeoutError` error will be raised, connection will be reconnected and then the topic will be subscribe again. 
You can add judgement on message `type` field to avoid this problem, or you can add `timeout` in upper layer.

**Comment by penguinol** [comment]
I'm wondering whether re-connecting and re-subscribing will lead to losing of notify messages which are sending at that moment .

**Comment by NoneGG** [comment]
@penguinol Yes, if message is published before topic is successfully subscribed, the message will be missed.

**Comment by NoneGG** [comment]
You are right, re-connection and re-subscription seems needless here

**Comment by NoneGG** [comment]
@penguinol 
After debug, i find out that connection is disconnected when handling `CancelledError` raise in coroutine caused by timeout, and then re-connection is brought out. 
To see the reason of this change, you can see #56 

Now a check will be placed before disconnection, when `CancelledError` is raised, only connection with not empty buffer will be disconnected. Useless re-connection and re-subscription will be leaved out.
